tkr/controller/tkr-source/fetcher/fetcher.go

```diff
@@ -176,6 +176,9 @@ func (f *Fetcher) saveTKRCompatibilityCM(ctx context.Context, ns string, metadat
 }
 
 func (f *Fetcher) fetchCompatibilityMetadata() (*tkrv1.CompatibilityMetadata, error) {
+	if f.Config.BOMMetadataImagePath == "" {
+		return nil, errors.New("compatibility metadata image path should not be empty")
+	}
```

Comment:
Can we insert this check _before_ starting the controller? It feels like, at this point in the code, it's already too late.

So, maybe in main.go where we just got these config values and are creating Config objects.

Reply:
sure. updated